### PR TITLE
Implement 'version` sub-command

### LINKED ISSUE #409: Add version subcommands to clusterawsadm and clusterctl binaries
/kind feature

**Describe the solution you'd like**
Currently there is no way to know if the versions of clusterctl or clusterawsadm used are what you expect them to be (or to compare the version installed to the versions released). There should be a version subcommand for each binary to output the version.

### PR DESCRIPTION
**Pull request by ashish-amarnath**
implement 'version` sub-command:

* create a sharable versioninfo package
* add version command to `clusterctl` and `clusterawsadm`
* version command will have the following details, as the version info:
** git branch
** git commit
** release tag
** Git major and minor version
** git tree status
Signed-off-by: Ashish Amarnath <ashish.amarnath@gmail.com>

**What this PR does / why we need it**:
This PR implements `version` sub-command for `clusterawsadm` and `clusterctl`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #409 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._ NO

**Release note**:

```release-note
1. `clusterawsadm` now has a `version` subcommand that exposes the status of the repo from which `clusterawsadm` was built/ released.

2. `clusterctl` now has a `version` subcommand that exposes the status of the repo from which `clusterctl` was built/released.
```

cc @detiber @stevesloka

### CODE REVIEW

**Comment by randomvariable** [comment]
I'll take a look tomorrow.

/ok-to-test

**Comment by detiber** [comment]
@chuckha I think for now we need to bundle clusterawsadm with a release, unless we have a better story around managing the pre-reqs needed.

File: cmd/versioninfo/versioninfo.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+func VersionCmd() *cobra.Command { // nolint
+	return &cobra.Command{
+		Use:   "version",
+		Short: "Print version of this binary",
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Printf("Git Branch: %q\n", GitBranch)
+			fmt.Printf("Git commit: %q\n", GitCommit)
+			fmt.Printf("Release tag: %q\n", ReleaseTag)
+			fmt.Printf("Git tree state: %q\n", GitTreeState)
+			fmt.Printf("Git version: Major:%q, Minor:%q\n", GitMajor, GitMinor)
```
Comment (detiber) [inline cmd/versioninfo/versioninfo.go]: This output appears misleading for alpha releases (such as the current one).

I'd like to see the output changed up to:

- Have a simplified output by default
  - If git tree state is clean and release tag commit is the current commit, print the tagged version
  - If git tree state is dirty or the release tag commit is not the current commit, output a modified simple version
- An optional verbose output that gave more details about the version, such as the current output

Comment (ashish-amarnath) [reply cmd/versioninfo/versioninfo.go]: Here's what the version output looks like:
```
$ clusterawsadm version
Version Info: GitReleaseTag: "v1.0.0-alpha.3", MajorVersion: "1", MinorVersion:"0", GitReleaseCommit:"1d5a033234f40b", GitTreeState:"dirty"
$ clusterawsadm version -l
Version Info:
GitReleaseTag: "v1.0.0-alpha.3", Major: "1", Minor: "0", GitRelaseCommit: "1d5a033234f40b"
Git Branch: "version-cmd"
Git commit: "b18ba239425057-dirty"
Git tree state: "dirty"
$ clusterawsadm version --long
Version Info:
GitReleaseTag: "v1.0.0-alpha.3", Major: "1", Minor: "0", GitRelaseCommit: "1d5a033234f40b"
Git Branch: "version-cmd"
Git commit: "b18ba239425057-dirty"
Git tree state: "dirty"
$ clusterawsadm version foo
Error: accepts 0 arg(s), received 1
Usage:
  clusterawsadm version [flags]

Flags:
  -h, --help   help for version
  -l, --long   Print longhand version info

accepts 0 arg(s), received 1
```

File: cmd/versioninfo/versioninfo.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+func printVerboseVersionInfo() {
+	fmt.Println("Version Info:")
+	fmt.Printf("GitReleaseTag: %q, Major: %q, Minor: %q, GitRelaseCommit: %q\n", GitReleaseTag, GitMajor, GitMinor, GitReleaseCommit)
+	fmt.Printf("Git Branch: %q\n", GitBranch)
+	fmt.Printf("Git commit: %q\n", GitCommit)
+	fmt.Printf("Git tree state: %q\n", GitTreeState)
+}
+
```
[the remark this answers is not in the document]
Comment (ashish-amarnath) [reply cmd/versioninfo/versioninfo.go]: I used that as a guideline. But @detiber's suggestion of controlling details based on the repo state made more sense to me. Just from a developer's perspective.

File: cmd/versioninfo/versioninfo.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
+package versioninfo
```
[the remark this answers is not in the document]
Comment (ashish-amarnath) [reply cmd/versioninfo/versioninfo.go]: agree. I'll create an issue for it.

[the remark this answers is not in the document]
Comment (ashish-amarnath) [reply cmd/versioninfo/versioninfo.go]: @detiber @vincepri created https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/441
Fixing the headers now.

File: cmd/versioninfo/versioninfo_test.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package versioninfo
```
[the remark this answers is not in the document]
Comment (ashish-amarnath) [reply cmd/versioninfo/versioninfo_test.go]: fixed.

File: cmd/versioninfo/versioninfo.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashish-amarnath) [reply cmd/versioninfo/versioninfo.go]: fixed

**Pull request merged**